freyja_examples/src/temporal_provider.cpp

```diff
@@ -9,30 +9,69 @@
   
   -- aj / 23rd Nov, 2017.
 */
-#include <cmath>
-#include <ros/ros.h>
-#include <std_msgs/UInt8.h>
-#include <freyja_msgs/ReferenceState.h>
+
+#include "temporal_provider.h"
 
 #define ROS_NODE_NAME "trajectory_provider"
-typedef freyja_msgs::ReferenceState TrajRef;
+#define UPDATE_RATE 50														// update rate of the trajectory (Hz)
+
+Temporal_Traj::Temporal_Traj() : Node( ROS_NODE_NAME )
+{
+	init_time = now();
+
+	declare_parameter<std::string>("traj_type", traj_type);
+	declare_parameter<int>("agg_level", agg_level);
 
-#define DEG2RAD(D) ((D)*3.1415326/180.0)
+	get_parameter("traj_type", traj_type);
+	get_parameter("agg_level", agg_level);
 
-// global decl for "start time". This can be reset by a callback
-ros::Time init_time;
+	time_reset_sub_ = create_subscription<std_msgs::msg::UInt8>(
+	"/reset_trajectory_time", 1, std::bind(&Temporal_Traj::timer_reset_cb, this, _1));
+
+	traj_pub_ = create_publisher<TrajRef>("/reference_state", 1);
+
+	traj_update_timer_ = create_wall_timer(
+	20ms, std::bind(&Temporal_Traj::traj_update_cb, this));
+}
 
-void timeResetCallback( const std_msgs::UInt8::ConstPtr &msg )
+void Temporal_Traj::timer_reset_cb( std_msgs::msg::UInt8::SharedPtr msg )
 {
-  if( msg -> data == 1 )
-  {
-    init_time = ros::Time::now();
-    ROS_WARN( "%s: Time reset requested!", ROS_NODE_NAME );
-  }
+	if (msg->data == 1)
+	{
+		init_time = now();
+		RCLCPP_WARN(get_logger(), "%s: Time reset requested!", ROS_NODE_NAME);
+	}
 }
 
-// HOVER AT A POINT 
-TrajRef getHoverReference( const ros::Duration &cur_time )
+void Temporal_Traj::traj_update_cb()
+{
+	get_parameter("traj_type", traj_type);
```

Comment:
I'm not sure that this is essential (esp. for the purposes of an example). Resetting this parameter on the fly will lead to discontinuous and somewhat unexpected trajectories mid-flight.